metadata/src/main/java/org/apache/kafka/controller/ConfigurationControlManager.java

```diff
@@ -352,6 +367,71 @@ private ApiError validateAlterConfig(
         return ApiError.NONE;
     }
 
+    List<ApiMessageAndVersion> maybeTriggerPartitionUpdateOnMinIsrChange(List<ApiMessageAndVersion> records) {
```

Comment:
As we discussed before, this will generate way too many records to be practical in many cases.

Rather than doing this, how about having a new record type that clears the ELRs of all partitions? Then we can have the controller generate that whenever the cluster-level min.insync.replicas is changed.